cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -150,3 +206,56 @@ func (crs *containerRegistryService) createRegistriesClient(
 
 	return client, nil
 }
+
+func (crs *containerRegistryService) getAcrToken(
+	ctx context.Context,
+	subscriptionId string,
+	loginServer string,
+) (*acrToken, error) {
+	creds, err := crs.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("getting credentials for subscription '%s': %w", subscriptionId, err)
+	}
+
+	token, err := creds.GetToken(ctx, policy.TokenRequestOptions{Scopes: auth.LoginScopes})
```

Comment:
Logically, I think using `azure.ManagementScope` makes more sense here over `auth.LoginScopes`. `auth.LoginScopes` are the default scopes we request to login, whereas here we want to explicitly use the scope that is required for the token exchange.